addons/website_sale_picking/static/tests/tours/onsite_payment_tour.js

```diff
@@ -1,97 +1,80 @@
-    }, {
-        content: 'Head back to main page',
-        trigger: '.nav-link:contains("Shop")'
+        test: true,
+        url: '/shop?&search=Chair floor protection'
     },
+    [
+        wTourUtils.clickOnElement('Select the first product', 'a:contains(Chair floor protection)'),
+        wTourUtils.clickOnElement('Add to cart', '#add_to_cart'),
+        wsTourUtils.goToCart(),
+        wTourUtils.clickOnElement('Proceed to checkout', 'a:contains(Process Checkout)'),
+        ...fillAdressForm(),
 
+        wTourUtils.clickOnElement('Example shipping On Site', '.o_delivery_carrier_select:contains("Example shipping On Site")'),
+        wTourUtils.clickOnElement('pay button', 'button[name="o_payment_submit_button"]:visible:not(:disabled)'),
+        {
+            content: "Check if the payment is successful",
+            trigger: 'p:contains(Your payment has been successfully processed but is waiting for approval.)',
+        },
+        wTourUtils.clickOnElement('Shop', 'a:contains("Shop")'),
 
-    { // Test multi products (Either physical or not)
-        content: 'Select the first product',
-        trigger: 'a:contains("Customizable Desk")',
-    }, {
-        content: 'Add the product to the cart',
-        trigger: 'a:contains("ADD TO CART")'
-    }, {
-        content: 'Validate variant choice',
-        trigger: 'button:contains("Continue Shopping")'
-    }, {
-        content: 'Head back to main page',
-        trigger: '.nav-link:contains("Shop")'
-    }, {
-        content: 'Select the second product (Non physical)',
-        trigger: 'a:contains("Warranty")',
-    }, {
-        content: 'Add the product to the cart',
-        trigger: 'a:contains("ADD TO CART")'
-    }, {
-        content: 'Go to cart',
-        trigger: '.nav-link[href="/shop/cart"]:contains("2")'
-    }, {
-        content: 'Go to payment page',
-        trigger: 'a:contains("Process Checkout")'
-    }, {
-        content: 'Click on onsite delivery carrier',
-        trigger: '.o_delivery_carrier_select:contains("On Site")'
-    }, {
-        content: 'Click on on site payment provider',
-        trigger: '.o_payment_option_card:contains("Pay in store when picking the product")'
-    }, {
-        content: 'Click the pay button',
-        trigger: 'button[name="o_payment_submit_button"]:visible:not(:disabled)'
-    }, {
-        content: 'Await the confirmation page',
-        trigger: 'td:contains("Pay in store when picking the product")'
-    }, {
-        content: 'Head back to main page',
-        trigger: '.nav-link:contains("Shop")'
-    },
-    // Test without any physical product (option pay on site should not appear)
+        // Test multi products (Either physical or not)
+        ...wsTourUtils.searchProduct("Customizable Desk"),
+        wTourUtils.clickOnElement('Select the first product', 'a:contains(Customizable Desk)'),
+        wTourUtils.clickOnElement('Add to cart', '#add_to_cart'),
```

Comment:
Since we are adding utils function, what about an utils that does the search and add to cart in one go?
(we can keep searchProduct) but maybe something like 
```js
function addToCart(product, search = true) {
    const steps = [];
    if (search) {
        steps.push(...searchProduct(product));
    }
    steps.push(...[select product, add to cart]);
    return steps;
}
```